### PR TITLE
React jwt

### PR DESCRIPTION
**Pull request by flcarre**
related to #790

### CODE REVIEW

**Comment by flcarre** [comment]
remaining

- [ ] add react jwt to generator
- [ ] modify app.tsx by adding <LoginForm /> composant
- [ ] add tests to generated
- [ ] add tests to generator

File: src/main/resources/generator/client/react/src/main/webapp/app/login/primary/LoginForm/index.tsx.mustache [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+export const UserInfoContext = createContext<UserInfoContextType | Record<string, never>>({});
+
+const LoginForm = () => {
+const [isOpen, setOpen] = useState(false);
+const [username, setUsername] = useState<any>('');
+  const [token, setToken] = useState<any>('');
+
+    const userInfoContextValues = {
+    setUsername,
```
[the remark this answers is not in the document]
Comment (fdelbrayelle) [reply src/main/resources/generator/client/react/src/main/webapp/app/login/primary/LoginForm/index.tsx.mustache]: I mean in the generated project

File: src/main/resources/generator/client/react/src/main/webapp/app/common/primary/app/LoginApp.tsx.mustache [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import LoginForm from '@/login/primary/loginForm';
+
+import './App.css';
+
+import React from 'react';
+
+function App() {
+  return (
+    <div className="App">
```
Comment (fdelbrayelle) [inline src/main/resources/generator/client/react/src/main/webapp/app/common/primary/app/LoginApp.tsx.mustache:9]: Why is there just a basic template here and no login form?

File: src/main/resources/generator/client/react/src/main/webapp/app/login/primary/loginForm/index.tsx.mustache [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  const [username, setUsername] = useState<any>('');
+  const [token, setToken] = useState<any>('');
+
+  const userInfoContextValues = {
+    setUsername,
+    setToken,
+  };
+
+  const onClickLoginButton = () => {
+    setOpen(true);
```
Comment (fdelbrayelle) [inline src/main/resources/generator/client/react/src/main/webapp/app/login/primary/loginForm/index.tsx.mustache:19]: Oneline ?

File: src/main/resources/generator/client/react/src/main/webapp/app/login/primary/loginForm/index.tsx.mustache [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  };
+
+  const onCloseModal = () => {
+    setOpen(false);
```
Comment (fdelbrayelle) [inline src/main/resources/generator/client/react/src/main/webapp/app/login/primary/loginForm/index.tsx.mustache:23]: Oneline ?

File: src/main/resources/generator/client/react/src/main/webapp/app/login/primary/loginForm/index.tsx.mustache [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  };
+
+  return (
+    <div>
+      <Button auto shadow onClick={onClickLoginButton}>
+        Se connecter
```
Comment (fdelbrayelle) [inline src/main/resources/generator/client/react/src/main/webapp/app/login/primary/loginForm/index.tsx.mustache:29]: No i18n ?

File: src/main/resources/generator/client/react/src/main/webapp/app/login/primary/loginModal/index.tsx.mustache [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+import './LoginModal.scss';
+
+const LoginModal = ({ open, onClose }: LoginModalType) => {
+const { register, handleSubmit } = useForm();
+const [error, setError] = useState<boolean>(false);
+  const { setUsername, setToken } = useContext(UserInfoContext);
+
+  const onSubmit = (loginData: any) => {
+  if (loginData.username && loginData.password) {
+  login({ ...loginData, setUsername, setToken });
```
Comment (fdelbrayelle) [inline src/main/resources/generator/client/react/src/main/webapp/app/login/primary/loginModal/index.tsx.mustache:17]: `npm run prettier:format` on generated project to indent better mustache files

File: src/main/resources/generator/client/react/src/main/webapp/app/login/primary/loginModal/index.tsx.mustache [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  setError(false);
+  onClose();
+  } else setError(true);
+  };
+
+  const handleClose = () => {
+  setError(false);
```
Comment (fdelbrayelle) [inline src/main/resources/generator/client/react/src/main/webapp/app/login/primary/loginModal/index.tsx.mustache:24]: same : indentation :warning:

**Review by fdelbrayelle — CHANGES_REQUESTED** [review]

File: src/main/resources/generator/client/react/src/main/webapp/app/login/primary/loginForm/index.tsx.mustache (hunk 5 above)
Comment (flcarre) [reply src/main/resources/generator/client/react/src/main/webapp/app/login/primary/loginForm/index.tsx.mustache:29]: I18n not already plug

**Review by fdelbrayelle — DISMISSED** [review]

**Comment by pascalgrimaud** [comment]
Can you solve the conflict plz? So it can be merged

**Comment by flcarre** [comment]
> Can you solve the conflict plz? So it can be merged

Done

**Pull request merged**